flopy/discretization/modeltime.py

```diff
@@ -7,61 +12,211 @@ class ModelTime:
+        """
         return self._start_datetime
 
     @property
     def perlen(self):
-        return self._period_data["perlen"]
+        """
+        Returns a list or array of stress-period lengths
+
+        """
+        return self._perlen.copy()
 
     @property
     def nper(self):
-        return len(self._period_data["perlen"])
+        """
+        Returns the number of stress periods
+        """
+        return len(self._perlen)
 
     @property
     def nstp(self):
-        return self._period_data["nstp"]
+        """
+        Returns a list or array of number of time steps per stress period
+        """
+        return self._nstp.copy()
 
     @property
     def tsmult(self):
-        return self._period_data["tsmult"]
+        """
+        Returns the time step multiplier value for each stress period
+
+        """
+        return self._tsmult.copy()
+
+    @property
+    def period_data(self):
+        """
+        Returns a tuple of period data for the MF6 TDIS package containing records
+        of [(perlen, nstp, tsmult), ....] for each stress period
+        """
+        return [(per, self.nstp[ix], self.tsmult[ix]) for ix, per in self.perlen]
 
     @property
     def steady_state(self):
+        """
+        Returns a boolean that indicates either steady-state or transient stress period
+
+        """
         return self._steady_state
 
     @property
     def totim(self):
+        """
+        Returns a list of totim values at the end of each time-step
+
+        """
+        if not self._totim_dict:
+            self._set_totim_dict()
+
+        return list(self._totim_dict.values())
+
+    @property
+    def kper_kstp(self):
+        """
+        Returns a list of kper, kstp tuples that correspond to totim
+
+        """
+        if not self._totim_dict:
+            self._set_totim_dict()
+        return list(self._totim_dict.keys())
+
+    @property
+    def tslen(self):
+        """
+        Method to get a list of time step lengths for the entire model period
+
+        """
+        n = 0
+        tslen = []
+        totim = self.totim
+        for ix, stp in enumerate(self.nstp):
+            for i in range(stp):
+                if not tslen:
+                    tslen = [totim[n]]
+                else:
+                    tslen.append(totim[n] - totim[n - 1])
+                n += 1
+
+        return np.array(tslen)
+
+    def get_datetime_string(self, datetime_obj):
+        """
+        Method to get a standarized ISO 8601 compliant datetime string
+
+        Parameters
+        ----------
+        datetime_obj : various objects
+            user supplied datetime representation. Please see the
+            ModelTime.datetime_from_user_input documentation for a list
+            of the supported representation types
+
+        """
+        dt = self.datetime_from_user_input(datetime_obj)
+        return dt.strftime("%Y-%m-%dT%H:%M:%S")
+
+    def set_start_datetime(self, datetime_obj):
```

Comment:
Could this be a property setter, instead of a normal method?

Reply:
Changed to a property setter